systemverilog-plugin/UhdmAst.cc

```diff
@@ -2173,13 +2174,64 @@ void UhdmAst::process_typespec_member()
     add_multirange_wire(current_node, packed_ranges, unpacked_ranges);
 }
 
-void UhdmAst::process_enum_typespec()
+void UhdmAst::process_enum_typespec(const UHDM::BaseClass *object)
 {
     // BaseTypespec specifies underlying type of the enum.
     // The BaseTypespec has at most one explicit packed dimension (range).
     // When base type is not specified in SystemVerilog code, it is assumed to be an int.
     // Type of enum items (constants) is the same as the enum type.
     current_node = make_ast_node(AST::AST_ENUM);
+
+    const auto *enum_object = object->Cast<const UHDM::enum_typespec *>();
+    const auto *typespec = enum_object->Base_typespec();
+
+    if (typespec && typespec->UhdmType() == UHDM::uhdmlogic_typespec) {
+        // If it's a logic_typespec, try to reduce expressions inside of it.
+
+        const auto *logic_typespec_obj = enum_object->Base_typespec()->Cast<const UHDM::logic_typespec *>();
+        std::vector<UHDM::range *> ranges;
+        // Check if ranges exist.
+        if (logic_typespec_obj->Ranges()) {
+            ranges = logic_typespec_obj->Ranges()[0];
+        }
+        for (UHDM::range *range_obj : ranges) {
+            // For each range, take both left and right and reduce them if they're of type uhdmoperation.
+            const auto *leftrange_obj = range_obj->Left_expr();
+            const auto *rightrange_obj = range_obj->Right_expr();
+            log_assert(leftrange_obj);
+            log_assert(rightrange_obj);
+
+            if (leftrange_obj->UhdmType() == UHDM::uhdmoperation) {
```

Comment:
Lets extract this to separate function so we won't need to repeat same ``if`` twice.

Reply:
I've came up with 2 ideas:

1. We can have a wrapper to `reduceExpr` that will take generic objects and reduce them.
This won't allow to reduce the two `if` statements into one,
as left and right expressions are handled by different methods (`Left_expr` and `Right_expr`),
but if we handle more cases later, we'll be able to use the same function.

2. We can have a wrapper to `reduceExpr` that will specifically take a `UHDM::range *`.
With this, we can replace the two `if` statements with a single call to the function.
This function would be useful only for `UHDM::range *` so it won't be as generic as 1).

I've implemented 1) — I've extracted some of the code to a function but there are still two `if` statements.
I'm not really against 2) but I thought 1) would be better.